add docs page and API reference entry for Naive Bayes (#763)
dask/dask-ml: docs/source/naive-bayes.rst

> Naive Bayes
> =========================

Reviewer: ```suggestion
Naive Bayes
===========
```

.. currentmodule:: dask_ml.naive_bayes

.. autosummary::
   GaussianNB

See `the scikit-learn documentation <https://scikit-learn.org/stable/modules/naive_bayes.html>`_ for more information about Naive Bayes classifiers.

These follow the scikit-learn estimator API, and so can be dropped into
existing routines like grid search and pipelines, but are implemented

> externally with new, scalable algorithms and so can consume distributed dask

Reviewer: ```suggestion
externally with new, scalable algorithms and so can consume distributed Dask
```

> arrays and dataframes rather than just single-machine NumPy and Pandas arrays

Reviewer: ```suggestion
arrays and DataFrames rather than just in-memory NumPy and Pandas arrays
```

> and dataframes.

Reviewer: ```suggestion
and DataFrames.
```
